src/IO/ReadWriteBufferFromHTTP.cpp

```diff
@@ -808,6 +808,11 @@ std::optional<time_t> ReadWriteBufferFromHTTPBase<UpdatableSessionPtr>::tryGetLa
 template <typename UpdatableSessionPtr>
 HTTPFileInfo ReadWriteBufferFromHTTPBase<UpdatableSessionPtr>::getFileInfo()
 {
+    if (!settings.http_make_head_request)
+    {
+            return HTTPFileInfo{};
+    }
```

Comment:
```suggestion
    /// May be disabled in case the user knows in advance that the server doesn't support HEAD requests.
    /// Allows to avoid making unnecessary requests in such cases.
    if (!settings.http_make_head_request)
        return HTTPFileInfo{};
```